dyne/ukf.py

```diff
@@ -5,6 +5,24 @@
 from ._common import check_input_arrays, check_measurements
 
 
+def _unscented_transform(x, P, Q, alpha):
+    n_states = len(x)
+    n_noises = len(Q)
+    l, V = linalg.eigh(P)
+    # Due to numerical issues computed eigenvalues could be negative,
+    # here we set them to zero
+    l[l < 0] = 0
+    Sigma = alpha * np.sqrt(n_states + n_noises) * linalg.block_diag(
```

Comment:
Am I correct that forming the full matrix is not necessary here?

That is noise and state can be sampled independently?

It seems to be true. So I would do it independently and return 2 probe vectors.

Reply:
We should definitely use sampling with the augmented matrix, it's just a matter of implementation.

I think the implementation / API can be improved. I will add some comments about that.